crates/holochain_sqlite/src/sql/conductor/select_valid_cap_grant_for_cap_secret.sql

```diff
@@ -2,17 +2,19 @@ SELECT
   Entry.blob
 FROM
   Entry
+  INNER JOIN Action ON Action.author = ?2
+  AND Action.entry_hash = Entry.hash
 WHERE
   Entry.cap_secret = ?1
   AND (
     SELECT
-      COUNT(Action.hash)
+      COUNT(UpdateActions.hash)
     FROM
-      Action
+      Action AS UpdateActions
     WHERE
-      Action.author = ?2
+      UpdateActions.author = ?2
```

Comment:
@neonphog Is this condition even needed? Can a cap grant be updated/deleted by anyone other than its author? And if it can, I suppose it would be invalid, right?

Reply:
Is it causing a slowdown, such that you want to remove it? The edge-case I'm thinking of is where the `original_entry_hash` is somehow the same across different originating agents, which may not even be possible depending on what gets hashed for cap entries, which I'm not clear on.

Reply:
I haven't checked whether it's causing a slowdown but would think it doesn't. It was because if it can be omitted it would make the query lexically simpler. Let's leave it in.